Grievance_Backend/grievance/src/main/java/com/grievance/controller/DepartmentController.java

```diff
@@ -57,14 +58,18 @@ public class DepartmentController {
 //  }
   @PostMapping("/save")
   public ResponseEntity<?> saveDepartment(
-    @RequestParam final String email,
-    @RequestParam final String password,
+    @RequestHeader final String email,
+    @RequestHeader final String password,
     @RequestBody final DepartmentInDto departmentInDto
   ) {
       Boolean boolean1 = authenticatingUser.checkIfUserIsAdmin(email, password);
+      Optional<DepartmentOutDto> optional = null;
       if (boolean1) {
-          Optional<DepartmentOutDto> optional =
-                 departmentService.saveDepartment(departmentInDto);
+          try {
+             optional = departmentService.saveDepartment(departmentInDto);
```

Comment:
where we are checking does optional is null or not , we do have functions like isPresent?

Reply:
I will update the code and will improve this area.